### ISSUE TITLE
Rework API and databases for multiple trees (e.g. extinct taxa)

### ISSUE THREAD

**Issue by hyanwong** [OneZoom/OZtree#700]
Here's a suggested way forward:

## app-config.ini
```
[db]
uri       = mysql://user:pass@127.0.0.1/OneZoom?set_encoding=utf8mb4
...
; adopt a convention that any appconfig starting with "db" describes a database
[dbtree_ExtantLife]
uri       = mysql://user:pass@127.0.0.1/ExtantLife?set_encoding=utf8mb4
...
[dbtree_ExtinctAmniotes]
uri       = mysql://user:pass@127.0.0.1/ExtantLife?set_encoding=utf8mb4
...
```

## static files
We rename `completetree_XXX.js` to `ExtantLife_tree.js`, `cut_position_map_XXX.js` to `ExtantLife_cutmap.js`, etc.

## db.py

We do

```python
myconf = AppConfig()
dbtree = {}
for k in myconf.keys():
    if k.startswith("db"):
        DALstring = myconf.take(f'{k}.uri')
        ...  # other stuff as in current db.py
        if k == "db":
            db = DAL(DALstring, ...)
        elif k.startswith("dbtree_"):
            dbname = k[len("dbtree_"):]
            dbtree[dbname] = DAL(DALstring, ...)
        else:
            raise ValueError("Unrecognisable database")
```

## controllers/API.py (and other controllers that use db.order_nodes or db.ordered_leaves)

instead of e.g.
```
query = db.ordered_leaves.ott == ott
```

use

```
tree_db_from_API = "ExtantLife"
query = dbtree[tree_db_from_API].ordered_leaves.ott == ott
```

Swapping these in to controllers other than API.py (and other modules) will force us to figure out what other parts of the OneZoom site are determined by the data in the tree (unfortunately there are a lot of cases!)

**Comment by davidebbo** [comment]
Thanks @hyanwong for starting this. There are really two distinct items here:
1. Using separate DBs for 'tree specific' tables and 'tree agnostic' tables
2. Adding support for distinct trees

The first can be done by splitting `db` into `db` and `dbtree` as you suggest, and then changing all relevant DB access code to use the right db. That should be relatively simple.

Once we have that, I would like to suggest an alternative to achieving multiple tree support. Instead of making the code base aware on 'n' distinct dbtree DBs, we can simply create different web2py applications for each tree. Specifically:
- All these apps would share identical files (e.g. via symlinks), with the exception of appconfig.ini
- Each appconfig.ini points at the same tree-agnostic `db`, but has its own `dbtree` DB
- Instead of needing our own logic to know which tree is being used for a given request, we get that entirely for free from the web2py routing that comes as a result of being in separate applications.

In other words, we just need to achieve the `db` vs `dbtree` DB separation, and we get multi-tree support essentially for free.

What do you think?

**Comment by hyanwong** [comment]
Thanks @davidebbo. I can definitely see some advantages in running on a separate app. However, there would be some hard-coding needed, as the front-page etc of the web site would need to know about the other apps (I guess that' possible, though).

Your version would mean running a different API for each tree, which is probably a neater way than reworking the API to allow different trees to be specified, so I like that. We could name the app "ExtinctAmniotes", I guess? Effort will be needed to turn off the required sections of the main app, but we've already committed to that, so perhaps we see how that goes first?

**Comment by jrosindell** [comment]
I'd vote for calling it ExtinctSpecies.  

Easy solution to the front page problem is the use the 'news' section which is in the DB and only link to the tree view of the other app from the menu which can be set up using a parameter in appconfig.ini

In the end this does make me wonder if one app just serves a tree (so we have two of these) and a third app serves the main page.  This moves us a step closer towards being able to offer easy to deploy containers for third parties to use with their own trees.

**Comment by davidebbo** [comment]
> In the end this does make me wonder if one app just serves a tree (so we have two of these) and a third app serves the main page

Yes, that is probably the best final state, and we can get there in stages. Does the main page require any DB access?
